internal/serialization/serialization.go

```diff
@@ -273,7 +301,63 @@ func (s *Service) registerIdentifiedFactories() error {
 	if err := s.registerSerializer(NewIdentifiedDataSerializableSerializer(factories)); err != nil {
 		return err
 	}
-	s.nameToID["identified"] = ConstantTypeDataSerializable
+	s.nameToID["identified"] = TypeDataSerializable
+	return nil
+}
+
+func (s *Service) lookupBuiltinSerializer(obj interface{}) pubserialization.Serializer {
+	switch obj.(type) {
+	case nil:
+		return nilSerializer
+	case bool:
+		return boolSerializer
+	case string:
+		return stringSerializer
+	case uint8:
+		return uint8Serializer
+	case uint16:
+		return uint16Serializer
+	case int:
+		return intSerializer
+	case int16:
+		return int16Serializer
+	case int32:
+		return int32Serializer
+	case int64:
+		return int64Serializer
+	case float32:
+		return float32Serializer
+	case float64:
+		return float64Serializer
+	case []bool:
+		return boolArraySerializer
+	case []string:
+		return stringArraySerializer
+	case []uint8:
+		return uint8ArraySerializer
+	case []uint16:
+		return uint16ArraySerializer
+	case []int:
+		return int64ArraySerializer
+	case []int16:
+		return int16ArraySerializer
+	case []int32:
+		return int32ArraySerializer
+	case []int64:
+		return int64ArraySerializer
+	case []float32:
+		return float32ArraySerializer
+	case []float64:
+		return float64ArraySerializer
+	case types.UUID:
+		return uuidSerializer
+	case time.Time:
+		return javaDateSerializer
+	case *big.Int:
```

Comment:
Why this is `*big.Int` but we have `time.Time` above?

Reply:
`time.Now` etc. time constructors return `time.Time` but `big.NewInt` constructor returns `*big.Int`.